packages/permissions/src/utils/has-permissions.ts

```diff
@@ -141,3 +172,71 @@ export const getInvalidPermissions = (
     isNil(actualPermissions[toCanCase(demandedPermission)])
   );
 };
+
+const hasDemandedDataFenceForStore = (
+  resourceToApplyDataFence: TResourceToApplyDataFence,
+  options: {
+    actualDataFence: TDataFenceByPermissionName;
+    actualDataFenceName: string;
+    demandedDataFence: TDemandedDataFence;
+  }
+): boolean => {
+  const hasDemandedPermission = hasPermission(options.demandedDataFence.name, {
```

Comment:
So `options.demandedDataFence.name` and `options.actualDataFenceName` are two different things?

Reply:
actualDataFenceName derives from `actualDataFences`, and `demanded*` from what was demanded

Reply:
Yeah but are they every different? Seems to me that `actualDataFenceName` is just the key while `options.actualDataFenceName.name` is the value which is always similar to the key?

Reply:
there is no `options.actualDataFenceName `, there is `options.actualDataFence`, maybe you misread the code or maybe I misunderstood you 😄 please elaborate more to help me understand this.